src/lib/utils/rpc-error-handler.ts

```diff
@@ -0,0 +1,59 @@
+import { ethers } from 'ethers';
+
+const notEnoughReward = (e: string) => e.includes('NotEnoughReward');
+const insufficientAllowance = (e: string) => e.includes('insufficient allowance');
+const lowSarcoBalance = (e: string) => e.includes('transfer amount exceeds balance');
+const sarcoDoesNotExist = (e: string) => e.includes('SarcophagusDoesNotExist');
+const sarcoNotCleanable = (e: string) => e.includes('SarcophagusNotCleanable');
+const sarcoIsActuallyUnwrappable = (e: string) => e.includes('SarcophagusIsUnwrappable');
+const notEnoughProof = (e: string) => e.includes('AccuseNotEnoughProof');
+const incorrectProof = (e: string) => e.includes('AccuseIncorrectProof');
+const invalidSignature = (e: string) => e.includes('InvalidSignature');
```

Comment:
These could be setup as a mapping or object -- not necessary to update, just an idea

```
const rewardError(e) {
  const a = e.indexOf('(') + 1;
  const b = e.indexOf(',');

  const available = e.substring(a, b);
  return `Not enough reward. Available: ${ethers.utils.formatEther(available)} SARCO`;
}

const ErrorMessageMapping = new Map<string, string | Function>([
[
 'NotEnoughReward',
 rewardError
], 
[
  'insufficient allowance', 
  'Insufficient allowance'
], 
[
  'transfer amount exceeds balance', 
  'Insufficient balance. Add some SARCO to your account to continue'
], 
...
)

export function handleRpcError(e: string) {
  for (let [key, value] of ErrorMessageMapping) {
    if (e.includes(key)) {
      if (typeof value === "function") {
        return value(e)
      } else {
        return value;
      }
    }
 }
}
```

Main reason for doing this would be to more tightly couple the error reason with the error message sent to users.

---

Separate note -- wondering if `RpcError` is the right name for this error type --- most (maybe all?) of these errors are revert errors from the contracts. I think of an RPC Error as being an issue with the RPC connection itself (i.e. Infura is down, or no RPC provider is set). This doesn't need to get changed now, but something to think about.

Reply:
Ah, very good points, both.